### PR TITLE
feat(cli): add get modules command in the cli

### LINKED ISSUE #196: Implement Cyclops CLI get methods
Implement `get` and `list` methods for modules

list - `cyctl get modules`
list output should look similarly to kubectl get
```bash
cyctl get modules

NAME     AGE
first    61s
second   46s
```

get - `cyctl get modules my-module`
get should return the requested module as YAML fetched from the cluster

You will also need to connect to the cluster and call the K8s API directly using .kubeconfig env variable

### PR DESCRIPTION
**Pull request by hanshal101**
## Description

In this PR I have added `get` command for `modules` this fetches the modules from the Kubernetes Cluster where Cyclops is deployed on. It fetches the `modules` from the tag:
- `"app.kubernetes.io/managed-by": "cyclops"`
- The output of the code looks something like this:
```
$ cyctl get modules
NAME       AGE
demo-app   15h34m38s
```
- And if the resource name after get is incorrect it will display a message: `Give the correct resource name`

## Screenshots/Images
image

### CODE REVIEW

File: cyctl/internal/get/modules.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+func ListModules(clientset *kubernetes.Clientset) {
+	labelSelector := metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app.kubernetes.io/managed-by": "cyclops",
+		},
+	}
+
+	modules, err := clientset.AppsV1().Deployments("").List(context.TODO(), metav1.ListOptions{
```
Comment (petar-cvit) [inline cyctl/internal/get/modules.go:20]: You are fetching for K8s deployments managed by Cyclops, but you will need to fetch Modules as CRDs.

I was looking into if you can use the [moduleset](https://github.com/cyclops-ui/cyclops/blob/f73ac60ec2ff796453536d5eace2a214a515e4a2/cyclops-ctrl/internal/cluster/v1alpha1/modules.go#L28) we already have and call for modules in a similar way you did with Deployments. I just need to refactor one part of cyclops-ctrl, and you can just import it and use it the same way it was used [here](https://github.com/cyclops-ui/cyclops/blob/aec912c35b5557eb2f72d5a08bf507e5dbbe5de2/cyclops-ctrl/internal/cluster/k8sclient/modules.go#L29)

File: cyctl/internal/get/modules.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	if err != nil {
+		if errors.IsNotFound(err) {
+			fmt.Println("No modules found.")
+			return
+		}
+		fmt.Printf("Error listing modules: %v\n", err)
+		return
+	}
+
+	fmt.Println("NAME       AGE")
```
Comment (petar-cvit) [inline cyctl/internal/get/modules.go:32]: I think we need a better way of formatting those columns. For longer deployment names, it's going to look messy.
Screenshot 2024-04-21 at 11 13 46

Comment (petar-cvit) [reply cyctl/internal/get/modules.go:32]: You can check this lib out https://github.com/ryanuber/columnize

**Review by petar-cvit — CHANGES_REQUESTED** [review]

**Comment by hanshal101** [comment]
Hey @petar-cvit I'm getting this error while installing the package:
```
PS H:\cyclops\cyctl> go mod tidy
go: finding module for package github.com/cyclops-ui/cyclops/cyclops-ctrl
go: github.com/cyclops-ui/cycops-cyctl/internal/get imports
        github.com/cyclops-ui/cyclops/cyclops-ctrl: module github.com/cyclops-ui/cyclops/cyclops-ctrl@latest found (v0.0.0-20240421163218-f48a78b7c0e7), but does not contain package github.com/cyclops-ui/cyclops/cyclops-ctrl
```

**Comment by petar-cvit** [comment]
Can you run `go mod tidy`

**Review by petar-cvit — APPROVED** [review]
Thanks @hanshal101

**Pull request merged**